### DISCUSSION TITLE
[firebase_messaging] Support mutable-content and Service App Extension (iOS)

### DISCUSSION THREAD

**Discussion by iosephmagno** [firebase/flutterfire#9317] [Feature request]
Hello everybody,
After facing issues with Silent Notifications (see #9300, #8277, #7874 and #6113) the community has realised that it's impossible to rely on iOS Silent Notifications when notifications must be displayed with high priority in all app states: foreground, background and terminated.

Main issue is with terminate state, because data-only iOS notifications head ups are not shown **if app is terminated AND app is not the most recent one opened on the device**, see #9300.

For above reason, data-only iOS notifications are not a viable solution for entire categories of apps such as chat apps and all apps at large that encrypt their content payload and need to decrypt it before displaying it in the notification head up.

To workaround this issue, it looks like we are forced to rely on remote notifications and Apple’s way to modify the content of a remote notification before it is displayed to users. Basically the solution requires to add **mutable-content:1** inside the notification payload and pass the content to a **notification Service App Extention**, which will then modify it before it gets displayed to user:
https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications

If you guys could work on it and give us guidance @russellwheatley @darshankawar, that would be really appreciated!

**Comment by karanianandkumar** [comment]
Is there any update on this issue @iosephmagno ? Did you find any other way around for this problem?
